test/utils/browser.js

```diff
@@ -81,6 +144,11 @@ exports.assertBrowser = function () {
   });
 };
 
+exports.simpleExecute = function (fn, args, cb) {
+  // DEV: Barebones execute function on the browser, this works when .execute does not
+  this.browser.execute(functionToString(fn).body, args, cb);
+};
```

Comment:
Ahhh, totally missed the `before` in `execute`. Nice catch. Having both a variation with and without `before` makes sense but having them be separate doesn't. I was going to suggest adding an `exports._execute` function that does the `slice` and `call` logic as `exports.execute` then use `exports._execute` in `exports.execute`.

However, I am realizing that `exports.execute` was being inconsistent with Selenium's API (which is what `simpleExecute` is using).

How about we alter the `exports.execute`/`exports.simpleExecute` to the following:

```js
exports._assertBrowser = function () {
  assert(this.browser, '`this.browser` is not defined. Please make sure `browserUtils.openMusic()` has been run.');
};

exports.assertBrowser = function () {
  before(exports._assertBrowser);
};

exports._execute = function (evalFn, args, cb) {
   exports._assertBrowser.call(this);
   this.browser.execute(functionToString(evalFn).body, args, cb);
};

exports.execute = function (evalFn, args) {
  before(exports.assertBrowser);
  before(function executeFn (done) {
    // Execute our function
    var that = this;
    exports._execute.call(this, evalFn, args, function handleResult (err, result) {
      // Save the result and callback
      that.result = result;
      done(err);
    });
  });
  after(function cleanup () {
    delete this.result;
  });
};
```